Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +16,143 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
+        }
+
+        public static int FindDigit(string equation)
+        {
+           return FixedMultiplication.FindDigit(equation);
+        }
+    }
+
+    enum EquationParts{
+        Operand1,
+        Operand2,
+        Result,
+        Invalid
+
+    }
+
+    class FixedMultiplication
+    {
+        
+        public static int FindDigit(string equation)
+        {
+           Equation ParsedEquation = ParseEquation(equation);
+           return ParsedEquation.CorrectEquationAndReturnMissingDigit();
+        }
+
+         private static Equation ParseEquation(string equation)
+        {
+            var regex = new Regex(@"(\d*\?*\d*)\*(\d*\?*\d*)=(\d*\?*\d*)", RegexOptions.Compiled);
+            var equationData = regex.Matches(equation);
+            if(equationData != null && equationData[0].Success)
+            {
+                var groups = equationData[0].Groups;
+                var operand1 = groups[1].Value;
+                var operand2 = groups[2].Value;
+                var result = groups[3].Value;
+
+ 
+
+                if (string.IsNullOrEmpty(operand1) || string.IsNullOrEmpty(operand2) || string.IsNullOrEmpty(result))
+                    return null;
+                return new Equation(operand1, operand2, "*", result);
+            }
+            return null;
+        }
+
+    }
+
+
+    class Equation{
+
+        public Equation(string operand1,string operand2,string operation,string result)
+        {
+            Operand1 = operand1;
+            Operand2 = operand2;
+            Operation = operation;
+            Result = result;
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public string Operand1 { get; private set; }
+        public string Operand2 { get; private set; }
+        public string Operation { get; private set; }
+        public string Result { get; private set; }
+
+        public int CorrectEquationAndReturnMissingDigit()
+        {
+            EquationParts needCorrection = GetEquationPartToBeCorrected();
+            string expectedValue;
+            bool validValue,validEquation;
+            switch(needCorrection)
+            {
+                case EquationParts.Operand1:
+                    expectedValue = Convert.ToString( int.Parse(Result) / int.Parse(Operand2) );

```

Comment:
Int32.Parse.
Always use actual classes while calling methods. 'int' is an alias.